### PR TITLE
3 commits: cleanup and fixes

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: bin/format_coverage.dart [hunk 1]
```diff
@@ -220,7 +220,8 @@ Environment parseArgs(List<String> arguments) {
 /// are contained by it if it is a directory, or a [List] containing the file if
 /// it is a file.
 List<File> filesToProcess(String absPath) {
-  final filePattern = RegExp(r'^dart-cov-\d+-\d+.json$');
+  print(absPath);
```
Comment (grouma) [inline bin/format_coverage.dart]: Remove?

File: lib/src/resolver.dart [hunk 2]
```diff
@@ -157,7 +157,9 @@ class Loader {
   /// Returns `null` if the resource could not be loaded.
   Future<List<String>> load(String path) async {
     try {
-      return File(path).readAsLines();
+      final lines = await File(path).readAsLines();
+      // ensure we await `readAsLines` within the try block so errors are caught
```
Comment (grouma) [inline lib/src/resolver.dart]: [nit] Capitalize and end in a period.

File: bin/format_coverage.dart [hunk 3]
```diff
@@ -220,7 +220,8 @@ Environment parseArgs(List<String> arguments) {
...
+  final filePattern = RegExp(r'^.*\.json$');
```
Comment (grouma) [inline bin/format_coverage.dart]: Add a simple test for this changed behavior?

File: lib/src/resolver.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/resolver.dart]: ack

File: bin/format_coverage.dart (hunk 1 above)
Comment (kevmoo) [reply bin/format_coverage.dart]: ack

File: bin/format_coverage.dart (hunk 3 above)
Comment (kevmoo) [reply bin/format_coverage.dart]: going to remove this commit and send another PR w/ something better here

File: lib/src/resolver.dart [hunk 4]
```diff
@@ -157,7 +157,9 @@ class Loader {
   /// Returns `null` if the resource could not be loaded.
   Future<List<String>> load(String path) async {
     try {
-      return File(path).readAsLines();
+      final lines = await File(path).readAsLines();
+      // Ensure `readAsLines` runs within the try block so errors are caught.
```
Comment (natebosch) [inline lib/src/resolver.dart]: I don't understand this comment

`return await File(path).readAsLines();` should be sufficient and clear. In a future SDK release the `await` won't be necessary.

File: test/chrome_test.dart [hunk 5]
```diff
@@ -28,18 +28,17 @@ Future<Uri> sourceUriProvider(String sourceUrl, String scriptId) async =>
 
 void main() {
   test('reports correctly', () async {
-    final preciseCoverage = json.decode(
+    final List preciseCoverage = json.decode(
```
Comment (natebosch) [inline test/chrome_test.dart]: This is not how we want to handle casts.

```dart
final preciseCoverage = json.decode(...) as List;
```

I'd prefer to fix the package analysis options than to expand use of implicit casts

File: lib/src/resolver.dart (hunk 4 above)
Comment (kevmoo) [reply lib/src/resolver.dart]: ack!

File: test/chrome_test.dart (hunk 5 above)
Comment (kevmoo) [reply test/chrome_test.dart]: Agreed. Let's do that in a follow-up. There are a lot of things to clean up.

**Pull request merged**